src/JuliusSweetland.OptiKey.Core/Models/XmlKeyboardModels/XmlVisualItems.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2019 OPTIKEY LTD (UK company number 11854839) - All Rights Reserved
+using System.Collections.Generic;
+using System.Reflection;
+using System.Xml.Serialization;
+using log4net;
+
+namespace JuliusSweetland.OptiKey.Models
+{
+    public class XmlVisualItem
```

Comment:
I'm a bit confused by this class. The filename is plural (XmlVisualItems, but the class singular). It then contains a set of Items. What exactly is being modelled here? Could we not have the XmlKeyboard with a set of Items, which are the XmlDynamicKeys/XmlScratchpad/XmlSuggestionRow/XmlSuggestionCol instead?

Reply:
XmlVisualItems is the new version of XmlKeys. I don't see an issue with merging it into XmlKeyboard.